src/modules/todo/components/todo/Todo.tsx

```diff
@@ -1,34 +1,47 @@
-import type { TodoItem } from './Todo.interface'
 import { useForm } from 'react-hook-form'
-import { useFetchTodos } from '../../../../hooks/useFetchTodos'
 import * as classes from './Todo.module.scss'
+import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query'
+import { fetchTodos, postTodo } from '../../api'
 
 type AddTodoFormValues = {
   text: string
 }
 
 function Todo() {
-  const [todos, setTodos] = useFetchTodos()
-
   const { register, handleSubmit, reset } = useForm<AddTodoFormValues>()
 
-  const onSubmit = handleSubmit((data) => {
-    fetch('/api/todos', { method: 'POST', body: JSON.stringify(data) })
-      .then(async (res) => {
-        const { todo } = (await res.json()) as { todo: TodoItem }
-        setTodos((prev) => [...prev, todo])
-        reset()
-      })
-      .catch((err) => {
-        console.error(err)
-      })
+  const queryClient = useQueryClient()
+
+  const { isLoading, data } = useQuery({
+    queryKey: ['todos'],
+    queryFn: fetchTodos,
+  })
+
+  const mutation = useMutation({
```

Comment:
It's better to put mutation and queries into separate hook, as you probably will use it a lot in a different components.